packages/botfuel-dialog/src/dialogs/prompt-dialog.js

```diff
@@ -236,24 +291,39 @@ class PromptDialog extends Dialog {
   async execute(adapter, userMessage, messageEntities) {
     logger.debug('execute', userMessage, messageEntities);
     const userId = userMessage.user;
-    const previouslyMatchedEntities =
-      (await this.brain.conversationGet(userId, this.parameters.namespace)) || {};
+
+    const dialogCache = await this.brain.conversationGet(userId, this.parameters.namespace);
+    const previouslyMatchedEntities = (dialogCache && dialogCache.entities) || {};
+    const previousQuestionEntity = (dialogCache && dialogCache.question) || undefined;
     logger.debug('execute: previouslyMatchedEntities', previouslyMatchedEntities);
+
     // Get missing entities and matched entities
     const { missingEntities, matchedEntities } = this.computeEntities(
```

Comment:
why can't we directly return missingEntities as a map?

Reply:
In fact I chose the lazy way (to not modify all the tests):
- leave computeEntities as it is (synchronised function), which return "object" missingEntities.
- afterward async computeQuestionEntities compute the final Map and SearchDialog only override this one:

Other alternative: make computeEntities async and return Map directly, and SearchDialog overrides this directly (which will call super.computeEntities).  

May be I should change the computeEntities directly? What do you think?

Reply:
I'll do that (change directly computeEntities and make it async), it is simpler and clearer.